### ISSUE TITLE
Loss Function Used

### ISSUE THREAD

**Issue by witnik** [DSPsleeporg/smiles-transformer#26]
Hi Shion,

I was trying to reproduce the model and realized that you used NLL loss in github yet it was mentioned in the paper to use Cross Entropy Loss. Would you mind suggesting which one of these loss function I should go further on my own?

Thanks

**Comment by shionhonda** [comment]
Cross entropy loss is equivalent to NLL loss following the log-softmax operation (see [here](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) for more detail).

My Transformer model includes log-softmax in the `forward` function, so it is correct to use NLL loss.
https://github.com/DSPsleeporg/smiles-transformer/blob/7ffa26e5bc68db0292164466398ff2edfa4a0f62/smiles_transformer/pretrain_trfm.py#L61

To answer your question, it's up to your implementation. If your model outputs log of probability, use  NLL loss. If it outputs logits, use cross entropy loss.
